homeassistant/components/pushbullet/sensor.py

```diff
@@ -177,6 +182,10 @@ async def async_added_to_hass(self) -> None:
             )
         )
 
+    async def async_will_remove_from_hass(self) -> None:
+        """Close running thread on remove."""
+        self.pb_provider.shutdown()
```

Comment:
We probably want to shutdown the thread once per config entry and not once per entity.

Reply:
Done in commit https://github.com/home-assistant/core/pull/74240/commits/fcbad139e2b251828e5c452f3e689712dfd77b07